File: day9-permissions-and-http/app/src/main/java/com/example/permissions/GameAdapter.java
package com.example.permissions;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.module.LoadMoreModule;
import com.chad.library.adapter.base.viewholder.BaseViewHolder;

import java.util.List;

public class GameAdapter extends BaseQuickAdapter<Game, BaseViewHolder> implements LoadMoreModule {

    private Context context;
    private List<Game> gameList;

    public GameAdapter(Context context, List<Game> gameList) {
        super(R.layout.item, gameList);
        this.context = context;
        this.gameList = gameList;
    }

    @Override
    protected void convert(BaseViewHolder holder, Game game) {
        ImageView gameImage = holder.getView(R.id.game_image);
        TextView gameName = holder.getView(R.id.game_name);
        TextView gameRating = holder.getView(R.id.game_rating);
        Button downloadButton = holder.getView(R.id.download_button);

        //gameImage.setImageResource(game.getGameImg());
        Glide.with(context)
                .load(game.getGameImg())
                .into(gameImage);
        gameName.setText(game.getGameName());
        gameRating.setText("Rating: " + game.getScore());

        downloadButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                // 处理下载按钮点击事件
            }
        });
    }

    @Override
    protected BaseViewHolder onCreateDefViewHolder(ViewGroup parent, int viewType) {
        return createBaseViewHolder(LayoutInflater.from(context).inflate(R.layout.item, parent, false));
    }
}

File: day11-kotlin/app/src/main/java/com/example/kotlin/RecyclerView/GameAdapter.kt
package com.example.kotlin.RecyclerView

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.module.LoadMoreModule
import com.chad.library.adapter.base.viewholder.BaseViewHolder
import com.example.kotlin.R

class GameAdapter(val adapterContext: Context, private var gameList: List<Game>) :
    BaseQuickAdapter<Game, BaseViewHolder>(R.layout.item, gameList.toMutableList()), LoadMoreModule {
    private var searchActivity: SearchActivity? = null

    interface OnItemClickListener {
        fun onItemClick(game: Game)
    }

    private var onItemClickListener: OnItemClickListener? = null
    fun setOnItemClickListener(listener: OnItemClickListener) {
        onItemClickListener = listener
    }

    fun setSearchActivity(activity: SearchActivity) {
        searchActivity = activity
    }


    override fun convert(holder: BaseViewHolder, game: Game) {
        val gameImage = holder.getView<ImageView>(R.id.game_image)
        val gameName = holder.getView<TextView>(R.id.item_text)
        val gameLike = holder.getView<ImageView>(R.id.like_image)

        //gameImage.setImageResource(game.getGameImg());
        gameImage.setImageResource(game.gameImg)
        gameName.text = game.gameName

        Glide.with(adapterContext).load(game.gameImg).into(gameImage)
        gameName.text = game.gameName
        if (game.like) {
            gameLike.setImageResource(R.drawable.like)
        } else {
            gameLike.setImageResource(R.drawable.nolike)
        }

        // 设置点击事件
        holder.itemView.setOnClickListener {
            onItemClickListener?.onItemClick(game)
        }

    }

    override fun onCreateDefViewHolder(parent: ViewGroup, viewType: Int): BaseViewHolder {
        return createBaseViewHolder(
            LayoutInflater.from(adapterContext).inflate(R.layout.item, parent, false)
        )
    }

    fun updateGameList(newGameList: List<Game>) {
        gameList = newGameList
    }

}